src/vec.rs

```diff
@@ -120,18 +124,64 @@ named!(range_literal <CompleteByteSlice, usize>, do_parse!(
 	(num * suffix)
 ));
 
+enum VectorRange {
+	// A bare range has no subquery associated with it
+	Bare(usize),
+	Subquery(usize, Option<usize>),
+}
+
+// Parse a range that must include a subrange selector. That is, the query's range component must
+// be of the form [<range>:] and may include an optional range after the colon. For example, [10m:]
+// and [1m:5m] are allowable, but a "bare" range like [10m] is not.
+fn subquery_range_selector(input: CompleteByteSlice) -> IResult<CompleteByteSlice, VectorRange> {
+	ws!(input,
+		delimited!(char!('['),
+			do_parse!(
+				range: range_literal >>
+				tag!(":") >>
+				resolution: opt!(range_literal) >>
+				( VectorRange::Subquery(range, resolution) )
+			),
+			char!(']')
+		)
+	)
+}
+
+// Parse a range vector selector. No subqueries are allowed here, it must be of the form specified
+// at https://prometheus.io/docs/prometheus/latest/querying/basics/#range-vector-selectors
+fn range_selector(input: CompleteByteSlice) -> IResult<CompleteByteSlice, VectorRange> {
+	do_parse!(input,
+				range: ws!(delimited!(
+								char!('['),
+								range_literal,
+								char!(']')
+							)) >>
+				( VectorRange::Bare(range) )
+	)
```

Comment:
Somehow you ended up with `\t\t\t` as a single indentation level here.

I guess I should just give up sprinkles of reasonably dense formatting and embrace `rustfmt` for everything.

Reply:
I just pushed rustfmt configuration according with reformatting existing code in advance. Sorry in advanced for inconvenience (although I guess not a lot from this pull request will end up in its next iteration.)

Reply:
No worries. My only defense is that the indent looked right in my editor? Running `rustfmt` at the end of the next revision will hopefully yield better results.